src/semiwrap/tool/update_yaml.py

```diff
@@ -0,0 +1,352 @@
+                    removals = diff[2]
+                    for removal in removals:
+                        # Make a patch that contains just this one removal
+                        modified_diff = [(diff[0], diff[1], [removal])]
+
+                        # Check if the removal is a full deletion of one of the keywords
+                        #
+                        # i.e. [('remove', 'classes.frc::Encoder', [('enums', {'IndexingType': None})])]
+                        # indicates that the enum block was completely removed
+                        if removal[0] in GENERATED_KEYWORDS:
+                            original = dictdiffer.patch(modified_diff, original)
+                            continue
+
+                        # Check if this removal is a sub-change of one of the keywords
+                        #
+                        # i.e. [('remove', 'classes.frc::Encoder.methods', [('PIDGet', {'rename': 'pidGet'})])]
+                        # indicates that the PIDGet function was removed, while other Encoder.methods remain.
+                        for item in GENERATED_KEYWORDS:
+                            if diff[1].endswith(item):
+                                original = dictdiffer.patch(modified_diff, original)
+                                break
+
+            # Patch all of the additions while being aware of if the file / class / etc has been marked as "ignore"
+            if not original.get("defaults", {}).get("ignore", False):
+                for addition in additions:
+                    original_contents = dictdiffer.utils.dot_lookup(
+                        original, addition[1]
+                    )
+                    if original_contents.get("ignore", False):
+                        continue
+                    original = dictdiffer.patch(additions, original)
+
+            output_file = output_directory / f.relative_to("semiwrap")
+
+            if write:
+                output_file.parent.mkdir(parents=True, exist_ok=True)
+
+                with open(output_file, "w") as f:
+                    yaml_.dump(original, f)
+            else:
+                # Original dir might be the same as the output dir, so we have to dump it into a string
+                # rather than risk writing the file and doing a simpler diff.
+                strbuff = StringIO()
+                yaml_.dump(original, strbuff)
+                strbuff.seek(0)
+                merged_lines = strbuff.readlines()
+
+                with open(output_file, "r") as of:
+                    output_lines = of.readlines()
+
+                differences = difflib.unified_diff(
+                    output_lines,
+                    merged_lines,
+                    fromfile=str(f),
+                    tofile=str(f),
+                )
+                differences = list(differences)
+
+                if differences:
+                    is_dirty = True
+                    print("Diff for ", output_file)
```

Comment:
Final (probably) nit: if you run `semiwrap update-yaml --write` and its only updating files, it doesn't tell you that it did anything, which made me wonder if it broke.

I think printing out the diffs in both cases is probably the right thing to do here.